src/lib/components/Tabs/tabsContext.ts

```diff
@@ -0,0 +1,44 @@
+import { writable, get, type Writable } from 'svelte/store';
+import { setContext, getContext } from 'svelte';
+
+interface Tab {
+	value: string;
+}
+
+export interface TabsContext {
+	activeTab: Writable<string>;
+	tabs: Writable<Tab[]>;
+	registerTab: (tab: Tab) => void;
+	selectTab: (value: string) => void;
+}
+
+const createTabs = (): TabsContext => {
+	const activeTab = writable<string>('');
+	const tabs = writable<Tab[]>([]);
+
+	const registerTab = (tab: Tab) => {
+		tabs.update((currTabs) => [...currTabs, tab]);
+		if (get(activeTab) === null) activeTab.set(tab.value);
+	};
```

Comment:
I think the logic here is quite strange

`activeTab` is being intialized as empty string but later in `registerTab` function you check  `if get(activeTab) === null)`

since `activeTab` is never null, this condition will always be false.

Reply:
Oh my bad, it should be `get(activeTab) === ""` thanks!